File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/SellerController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbSeller;
import com.pinyougou.sellergoods.service.SellerService;
import entity.PageResult;
import entity.Result;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * controller
 *
 * @author Administrator
 */
@RestController
@RequestMapping("/seller")
public class SellerController {

    @Reference
    private SellerService sellerService;

    @RequestMapping("/findByPage")
    public PageResult<TbSeller> findByPage(@RequestBody TbSeller seller, int page, int size) throws Exception {
        if (page <= 0) {
            page = 1;
        }
        if (size <= 0) {
            size = 10;
        }
        return sellerService.findByPage(seller, page, size);
    }

    @RequestMapping("/findById")
    public TbSeller findById(String id) throws Exception {
        return sellerService.findById(id);
    }

    @RequestMapping("/updateStatus")
    public Result updateStatus(String id, String status) {
        try {
            sellerService.updateStatus(id, status);
            return new Result(true, "状态修改成功!");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "状态修改失败!");
        }
    }
}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/plugins/angularjs/controller/sellerController.js
//控制器
app.controller('sellerController', function ($scope, $controller, sellerService) {

    //继承baseController
    $controller('baseController', {$scope: $scope});

    //查询参数初始化
    $scope.searchEntity = {};

    //分页请求方法
    $scope.findByPage = function (page, size) {
        sellerService.findByPage(page, size, $scope.searchEntity).success(function (data) {
            $scope.list = data.rows;
            $scope.paginationConf.totalItems = data.total;
        })
    };

    //修改回显数据
    $scope.findById = function (id) {
        sellerService.findById(id).success(function (data) {
            $scope.entity = data;
        })
    };

    //修改数据状态数据
    $scope.updateStatus = function (id, status) {
        sellerService.updateStatus(id, status).success(function (data) {
            if (data.success) {
                $scope.reloadPage();
            } else {
                alert(data.message);
            }
        })
    };
});